awx/ui_next/src/components/AppContainer/AppContainer.jsx

```diff
@@ -34,6 +38,45 @@ const PageHeader = styled(PFPageHeader)`
   }
 `;
 
+/**
+ * The useStorage hook integrates with the browser's localStorage api.
+ * It accepts a localStorage key as its only argument and returns a
+ * state variable and setter function for that state variable.
+ *
+ * This utility behaves much like the standard useState hook with some
+ * key differences:
+ *   1. You don't pass it an initial value. Instead, the provided key
+ *      is used to retrieve the initial value from local storage. If
+ *      the key doesn't exist in local storage, null is returned.
+ *   2. Behind the scenes, this hook registers an event listener with
+ *      the Web Storage api to establish a two-way binding between the
+ *      state variable and its corresponding local storage value. This
+ *      means that updates to the state variable with the setter
+ *      function will produce a corresponding update to the local
+ *      storage value and vice-versa.
+ *   3. When local storage is shared across browser tabs, the data
+ *      binding is also shared across browser tabs. This means that
+ *      updates to the state variable using the setter function on
+ *      one tab will also update the state variable on any other tab
+ *      using this hook with the same key and vice-versa.
+ */
+function useStorage(key) {
+  const [storageVal, setStorageVal] = useState(
+    window.localStorage.getItem(key)
```

Comment:
If later on we need to improve performance. We can evaluate the usage of `useState Lazy initialization` here. See: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

Reply:
:man_shrugging: Perhaps. Although I wouldn't expect a bottleneck to happen here as this is an initialization step and doesn't run _too_ frequently. 

To be clear, I don't expect performance issues to be of a high concern w/ this PR. The reason I brought it in my last comment was to point out where in the code we'd probably want to look first if it starts to be a concern in the future.